### PR TITLE
getCurrentUser must be called before any update/delete operations()

### PR DESCRIPTION
**Pull request by ssuvorov-fls**
fixes https://github.com/OHDSI/Atlas/issues/2359

### CODE REVIEW

**Comment by anton-abushkevich** [comment]
It works, but:
1. Open a source, uncheck daimon (for example, "temp"), save
2. Open the source, create same daimon again, save
3. Repeat 1 and 2 several times

Now check the database table "source_daimon" - there will be all created and deleted daimons with priority set to "-1". Question: may be it will be cleaner to reuse existing deleted daimon, when creating a new one? I.e. check if we already have a daimon in the source with daimon_type and if we do - then update its name and priority, not creating a new one.

**Comment by anton-abushkevich** [comment]
Also:

if multiple daimons with the same type is not allowed for one source, then it would be appropriate to add a unique constraint on two column: `UNIQUE (source_id, daimon_type)`.

**Comment by chrisknoll** [comment]
I think maybe a reason to not reuse is that it will also reuse the source_id value, which may then accidently associate results (ie: generation results) to a given source_id.

Maybe a solution is to set the source_key to NULL if you do a soft delete?

File: src/main/resources/db/migration/postgresql/V2.8.0.20201103171300__add_source_daimon_unique_constraint.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE ${ohdsiSchema}.source_daimon ADD CONSTRAINT un_source_daimon UNIQUE (source_id,daimon_type);
```
Comment (anton-abushkevich) [inline src/main/resources/db/migration/postgresql/V2.8.0.20201103171300__add_source_daimon_unique_constraint.sql:1]: Migration fails, if the table already has dublicates. First, have to delete those with priority = -1.

**Review by anton-abushkevich — CHANGES_REQUESTED** [review]

**Review by anton-abushkevich — APPROVED** [review]

**Comment by chrisknoll** [comment]
This migration failed on SqlServer:

SQL State  : S0001
Error Code : 102
Message    : Incorrect syntax near 'sd1'.
Location   : db/migration/sqlserver/V2.8.0.20201103171300__add_source_daimon_unique_constraint.sql (C:\Git\WebAPI\target\classes\db\migration\sqlserver\V2.8.0.20201103171300__add_source_daimon_unique_constraint.sql)

**Comment by ssuvorov-fls** [comment]
@chrisknoll Thanks a lot and sorry for mistake

**Pull request merged**